aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -431,6 +433,39 @@ protected boolean isValidIdentifier(TableIdentifier tableIdentifier) {
         IcebergToGlueConverter.isValidTableName(tableIdentifier.name());
   }
 
+  @Override
+  public org.apache.iceberg.Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
```

Comment:
Is it possible to do this more generically in `BaseMetastoreCatalog` similar to a regular `create`?

```java
    public Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
      TableOperations ops = newTableOps(identifier);
      if (ops.current() != null) {
        throw new AlreadyExistsException("Table already exists: %s", identifier);
      }

      FileIO io = ops.io();
      TableMetadata metadata = TableMetadataParser.read(io, metadataFileLocation);

      try {
        ops.commit(null, metadata);
      } catch (CommitFailedException ignored) {
        throw new AlreadyExistsException("Table was created concurrently: %s", identifier);
      }

      return new BaseTable(ops, fullTableName(name(), identifier));
    }
```

That will rewrite the metadata file rather than using it directly, but it seems like it would work in most cases.

Reply:
@jackye1995, what do you think about this suggestion?

Reply:
@rdblue I have considered this particular suggestion and implemented it under #5037, please do have a look at the implementation...